src/support/threads.cpp

```diff
@@ -145,6 +149,18 @@ size_t ThreadPool::getNumCores() {
   return 1;
 #else
   size_t num = std::max(1U, std::thread::hardware_concurrency());
+#ifdef __linux__
+  // On linux we can do better since we can get the number of CPU that are
+  // actually usable by the current process.
+  cpu_set_t cpu_set;
+  CPU_ZERO(&cpu_set);
+  if (sched_getaffinity(0, sizeof(cpu_set), &cpu_set) == 0) {
+    num = 0;
+    for (int i = 0; i < CPU_SETSIZE; i++) {
+      if (CPU_ISSET(i, &cpu_set)) num++;
```

Comment:
```suggestion
      if (CPU_ISSET(i, &cpu_set)) {
        num++;
      }
```

Reply:
Yes, but I don't think we do any build with logging do we?

Reply:
I think I can do it with `BYN_DEBUG` macro which is enabled in non-release builds.

Reply:
Do we have any way to running a test only linux?

Reply:
Hmm.. looks like we already have zero testing of `BINARYEN_CORES` so maybe we can leave testing as a followup?